### PR TITLE
Onboarding for FIWARE Catalogue

### PR DESCRIPTION
**Pull request by jason-fox**
This PR does the following:

* Moves the Dockerfile to the standard location
* Restructure existing Documentation
    * Tidy up the root README
    * Add badges and re-write lede
    * Move details to `docs` directory
* Add prettier, husky, remark and textlint
   * Run prettier over code base (correct whitespace errors)
*  Creates an initial setup for ReadtheDocs

You will still need to enable the RTD project, Travis, Coveralls etc, but this will tidy up the repository prior to onboarding as an **Incubated** Generic Enabler

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,250 +1,119 @@
-# sigfox-iotagent
+# FIWARE IoT Agent for Sigfox
 
-* [Overview](#overview)
-* [Installation](#installation)
-* [Usage](#usage)
-* [Data Format](#dataformat)
-* [Development Documentation](#development)
+[![FIWARE IoT Agents](https://nexus.lab.fiware.org/static/badges/chapters/iot-agents.svg)](https://www.fiware.org/developers/catalogue/)
+[![License: APGL](https://img.shields.io/github/license/telefonicaid/iotagent-json.svg)](https://opensource.org/licenses/AGPL-3.0)
```
Comment (fgalan) [inline README.md]: iotagent-json-> sigfox-iotagent

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            - "4041:4041"
+            - "7896:7896"
+        environment:
+            - "IOTA_CB_HOST=orion"
+            - "IOTA_CB_PORT=1026"
+            - "IOTA_NORTH_PORT=4041"
+            - "IOTA_REGISTRY_TYPE=mongodb"
+            - "IOTA_MONGO_HOST=mongo-db"
+            - "IOTA_MONGO_PORT=27017"
+            - "IOTA_MONGO_DB=iotagent-json"
```
Comment (fgalan) [inline docker/README.md]: This DB doesn't fit the IOTA name.

I guess there is some default in the config.js file. Let's use that.

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        depends_on:
+            - mongodb
+        expose:
+            - "4041"
+            - "7896"
+        ports:
+            - "4041:4041"
+            - "7896:7896"
+        environment:
+            - "IOTA_CB_HOST=orion"
```
Comment (fgalan) [inline docker/README.md]: Do you have checked all these env vars exists also for IOTA Sigfox? Specially with regards to the southbound part (e.g. not sure if in this case we have a IOTA_HTTP port or Sigfox implies a fixed port).

In addition, for env vars that exists, the default value should match the one used by this agent. For instance, I'm not sure that 4041 is the right default for Sigfox (I may be wrong, anyway).

**Comment by fgalan** [comment]
Travis is currently in fail status (I think due to a grammar error).

File: docker/README.md (hunk 3 above)
Comment (fgalan) [reply docker/README.md]: We have been checking on the code and it seems that appart from the env vars defined in IOTA Agents Library, the IOTA Sigfox implements the following variable: IOTA_SIGFOX_PORT (from https://github.com/telefonicaid/sigfox-iotagent/blob/master/lib/configService.js#L41).

CC: @AlvaroVega

File: docker/README.md (hunk 2 above)
Comment (fgalan) [reply docker/README.md]: See https://github.com/telefonicaid/sigfox-iotagent/blob/master/config.js#L146

File: README.md (hunk 1 above)
Comment (jason-fox) [reply README.md]: Fixed efc68c0

File: docker/README.md (hunk 2 above)
Comment (jason-fox) [reply docker/README.md]: Fixed efc68c0

File: docker/README.md (hunk 3 above)
Comment (jason-fox) [reply docker/README.md]: `ENV` altered to use and expose the SIGFOX port. Fixed efc68c0

**Comment by jason-fox** [comment]
> Travis is currently in fail status (I think due to a grammar error).

This is due to `write-good` complaining about the AGPL text - I've disabled the offending rule. Fixed efc68c0

File: docker/README.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+-   `IOTA_PROVIDER_URL` - URL passed to the Context Broker when commands are registered, used as a forwarding URL
+    location when the Context Broker issues a command to a device
+
+### Further Information
+
+The full set of overrides for the general parameters applicable to all IoT Agents are described in the Configuration
+section of the IoT Agent Library
+[Installation Guide](https://iotagent-node-lib.readthedocs.io/en/latest/installationguide/index.html#configuration).
+
+Further settings for IoT Agent for Sigfox itself - can be found in the IoT Agent for Sigfox
```
[the remark this answers is not in the document]
Comment (fgalan) [reply docker/README.md]: Fixed in efc68c0bd1717361c702f7386c04a6cccf961a19#diff-a489144702a723431811d56a3c44690a

File: docker/README.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+environment variables such as those shown below:
+
+-   `IOTA_CB_HOST` - Hostname of the context broker to update context
+-   `IOTA_CB_PORT` - Port that context broker listens on to update context
+-   `IOTA_NORTH_PORT` - Port used for configuring the IoT Agent and receiving context updates from the context broker
+-   `IOTA_REGISTRY_TYPE` - Whether to hold IoT device info in memory or in a database
+-   `IOTA_MONGO_HOST` - The hostname of MongoDB - used for holding device and service information
+-   `IOTA_MONGO_PORT` - The port that MongoDB is listening on
+-   `IOTA_MONGO_DB` - The name of the database used in MongoDB
+-   `IOTA_HTTP_PORT` - The port where the IoT Agent listens for IoT device traffic over HTTP
```
Comment (fgalan) [inline docker/README.md]: Needs to be changed to IOTA_SIGFOX_PORT

Comment (jason-fox) [reply docker/README.md]: Fixed 0332f65

**Comment by fgalan** [comment]
@jason-fox regarding the follow up actions:

* With regards to Coveralls, I have enabled the repository at https://coveralls.io/github/telefonicaid/sigfox-iotagent but I only see a blank page. However, the coveralls travis run seems to be working based in what I seen at https://travis-ci.org/telefonicaid/sigfox-iotagent/jobs/496025547:

imagen

* With regards to RTD, do you have the link to the howto from FIWARE documentation at hand? I don't remember all the details of the process so it would be very helpful.

**Comment by fgalan** [comment]
Please ignore the bulleta bout Coveralls. Now I can see the coverage badge and the link is working. Probably it was a temporal glitch. Sorry the noise.

**Comment by jason-fox** [comment]
`https://readthedocs.org/dashboard/` - **Import a Project**

On the settings mostly defaults, but check the following:


**Name**: `iotagent-sigfox`

**Documentation type**: `mkdocs`

You'll also need to set up analytics sometime. see https://fiware-requirements.readthedocs.io/en/latest/analytics_readthedocs/index.html

**Comment by fgalan** [comment]
I have set up the RTD project but I'm getting an ugly error when trying to compile (source: https://readthedocs.org/projects/iotagent-sigfox/builds/8620034/)

```
python /home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/bin/mkdocs build --clean --site-dir _build/html --config-file /home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/checkouts/latest/mkdocs.yml
INFO    -  Cleaning site directory 
INFO    -  Building documentation to directory: /home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/checkouts/latest/_build/html 
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/mkdocs/nav.py", line 337, in _follow
    raise StopIteration
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/bin/mkdocs", line 10, in <module>
    sys.exit(cli())
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/mkdocs/__main__.py", line 156, in build_command
    ), dirty=not clean)
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/mkdocs/commands/build.py", line 282, in build
    build_pages(config, dirty=dirty)
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/mkdocs/commands/build.py", line 213, in build_pages
    site_navigation = nav.SiteNavigation(config)
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/mkdocs/nav.py", line 44, in __init__
    config, self.url_context)
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/mkdocs/nav.py", line 392, in _generate_site_navigation
    config_line, url_context, config):
  File "/home/docs/checkouts/readthedocs.org/user_builds/iotagent-sigfox/envs/latest/lib/python3.7/site-packages/mkdocs/nav.py", line 356, in _follow
    for sub in _follow(path, url_context, config, header=header, title=next_cat_or_title):
RuntimeError: generator raised StopIteration
```

Maybe I have configured something wrongly...

Any clue?

**Comment by jason-fox** [comment]
@fgalan can you give me RTD admin rights to fix it? I can't see the set-up

**Comment by fgalan** [comment]
I finally find the cause of the problem. Python interpretar must be 2.x (by default the project was created with 3.x and that was causing the problem):

imagen

Documentation is here: https://iotagent-sigfox.readthedocs.io/en/latest/

**Pull request merged**